### ISSUE TITLE
Elgg is not PHP 5.6 compatible

### ISSUE THREAD

**Issue by jdalsem** [Elgg/Elgg#11224]
DBAL/Doctrine dependency requires PHP 7

Discovered this because Doctrine is using Throwable interface which is only usable in PHP7

We should lock the version in Elgg 2.3 and probably choose to require PHP 7 in Elgg 3.0

**Comment by hypeJunction** [comment]
I have been updating composer deps with `--ignore-platform-reqs` flag on 7.0 installations without issues so far

**Comment by jdalsem** [comment]
Not exactly what i mean. The Elgg composer lock file has now locked dependencies that REQUIRE php 7.. we probably should limit those dependencies to version that do not REQUIRE php 7, otherwise php5.6 users could run into trouble (as i have)

**Comment by jdalsem** [comment]
simple fix in 2.3 we should change the dbal requirement to `~2.5.0` instead of `~2.5` because in `2.6` they required php7

**Comment by juho-jaakkola** [comment]
Alternative approach could be to add this to Elgg's `composer.json`:

```json
  "config": {
    "platform": {
      "php": "5.6"
    }
  },
```
That way the PHP 5.6 requirement wouldn't be bind to a specific package, but the project itself.
